src/libAtomVM/nifs.c

```diff
@@ -3300,3 +3362,163 @@ static term nif_maps_next(Context *ctx, int argc, term argv[])
+    return term_from_float(z, ctx);
+}
+
+#define DEFINE_UNARY_MATH_OP(moniker)                                   \
+    static avm_float_t math_##moniker(avm_float_t x)                    \
+    {                                                                   \
+        return moniker(x);                                              \
+    }                                                                   \
+                                                                        \
+    static term nif_math_##moniker(Context *ctx, int argc, term argv[]) \
+    {                                                                   \
+        UNUSED(argc);                                                   \
+        VALIDATE_VALUE(argv[0], term_is_number);                        \
+        term t = math_unary_op(ctx, argv[0], math_##moniker);           \
+        if (term_is_atom(t)) {                                          \
+            RAISE_ERROR(t);                                             \
+        } else {                                                        \
+            return t;                                                   \
+        }                                                               \
+    }
+
+#define DEFINE_BINARY_MATH_OP(moniker)                                  \
+    static avm_float_t math_##moniker(avm_float_t x, avm_float_t y)     \
+    {                                                                   \
+        return moniker(x, y);                                           \
+    }                                                                   \
+                                                                        \
+    static term nif_math_##moniker(Context *ctx, int argc, term argv[]) \
+    {                                                                   \
+        UNUSED(argc);                                                   \
+        VALIDATE_VALUE(argv[0], term_is_number);                        \
+        VALIDATE_VALUE(argv[1], term_is_number);                        \
+        term t = math_binary_op(ctx, argv[0], argv[1], math_##moniker); \
+        if (term_is_atom(t)) {                                          \
+            RAISE_ERROR(t);                                             \
+        } else {                                                        \
+            return t;                                                   \
+        }                                                               \
+    }
+
+#else
+
+#define DEFINE_UNARY_MATH_OP(moniker)                                   \
+    static term nif_math_##moniker(Context *ctx, int argc, term argv[]) \
+    {                                                                   \
+        UNUSED(ctx);                                                    \
+        UNUSED(argv);                                                   \
+        RAISE_ERROR(BADARG_ATOM);                                       \
+}
```

Comment:
Could we align these the closing curly braces line 3499? We really should get rid of the clang-format exception list ;-)

Reply:
Done.  Thanks!